File: snippets/dart_snippets/lib/lnurl_withdraw.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';

Future<void> lnurlWithdraw() async {
  // ANCHOR: lnurl-withdraw
  /// Endpoint can also be of the form:
  /// lnurlw://domain.com/lnurl-withdraw?key=val
  String lnurlWithdrawUrl =
      "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4exctthd96xserjv9mn7um9wdekjmmw843xxwpexdnxzen9vgunsvfexq6rvdecx93rgdmyxcuxverrvcursenpxvukzv3c8qunsdecx33nzwpnvg6ryc3hv93nzvecxgcxgwp3h33lxk";

  InputType inputType = await breezSDKLiquid.instance!.parse(input: lnurlWithdrawUrl);
  if (inputType is InputType_LnUrlWithdraw) {
    BigInt amountMsat = inputType.data.minWithdrawable;
    LnUrlWithdrawRequest req = LnUrlWithdrawRequest(
      data: inputType.data,
      amountMsat: amountMsat,
      description: "<description>",
    );
    LnUrlWithdrawResult result = await breezSDKLiquid.instance!.lnurlWithdraw(req: req);
    print(result.data);
  }
  // ANCHOR_END: lnurl-withdraw
}

File: snippets/swift/BreezSDKExamples/Sources/LnurlWithdraw.swift
import BreezSDKLiquid
import Foundation

func withdraw(sdk: BindingLiquidSdk) -> LnUrlWithdrawResult? {
    // ANCHOR: lnurl-withdraw
    // Endpoint can also be of the form:
    // lnurlw://domain.com/lnurl-withdraw?key=val
    var response: LnUrlWithdrawResult?
    let lnurlWithdrawUrl = "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4exctthd96xserjv9mn7um9wdekjmmw843xxwpexdnxzen9vgunsvfexq6rvdecx93rgdmyxcuxverrvcursenpxvukzv3c8qunsdecx33nzwpnvg6ryc3hv93nzvecxgcxgwp3h33lxk"

    if let inputType = try? sdk.parse(input: lnurlWithdrawUrl){
        if case.lnUrlWithdraw(let `data`) = inputType {
            let amountMsat = data.maxWithdrawable
            let description = "Test withdraw"
            response = try? sdk.lnurlWithdraw(req: LnUrlWithdrawRequest(
                data: data,
                amountMsat: amountMsat,
                description: description))
        }
    }
    // ANCHOR_END: lnurl-withdraw
    return response
}
